tutorials/relay/implement_fma_transform.py

```diff
@@ -0,0 +1,141 @@
+"""How to use Relay to implement a simple two-operator fusion pass.
+==================================
+**Author**: `Jared Roesch <https://homes.cs.washington.edu/~jroesch/>`_
+
+In this tutorial, we will demonstrate how to write a fusion pass for
+the Relay IR. We demonstrate many Relay features including defining a
+new operator, a program transform, the NNVM compatibility layer,
+and executing the original and transformed programs on the Relay
+evaluator and TVM runtime system.
+"""
+
+################################################################
+# Introduction
+# -------------------------
+#
+# In this tutorial, we will demonstrate how to write a fusion pass for
+# the Relay IR. We demonstrate many Relay features including defining a
+# new operator, a program transform, the NNVM compatibility layer,
+# and executing the original and transformed programs on the Relay
+# evaluator and TVM runtime system.
+
+from typing import Any, Dict
+
+import numpy as np
+import tvm
+import topi
+
+from relay import ir, make as mk
+from relay.ir import OperatorId
+from relay.opt import ItemVisitor, ExprVisitor
+from relay.frontend.nnvm import Variable, symbol
+from relay.frontend.nnvm import compiler
+from relay.frontend.global_env import get_env
+from relay.operators.register import func_ty_to_placeholders, register_op
+from relay.eval import defn_to_pyfunc
+from relay.tyck import check_expr
+
```

Comment:
I don't see relay.frontend, eval, tyck etc in the current repo. Are they going to be upstreamed soon?

Reply:
This code still needs to be updated from the old branch, but hope is to upstream these pieces in the next few days. 

We have a few parallel efforts going on, some changes to the type checker for supporting VTA, the runtime compiler and evaluator, AD, debugging support, and work towards automatic quantization/8bit support. We will hopefully be releasing most of this in the next couple weeks. 

3 of our team members have been away for the summer so parallelization should pick up soon.